### PR TITLE
Microsoft storage cache api 2023 01 01

### PR DESCRIPTION
**Pull request by rebecca337**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month. **End of January 2023**
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month. **End of January 2023**
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by mentat9** [comment]
>               "$ref": "#/definitions/CloudError"

Non-blocking, but recommended that teams move to ErrorResponse from v4 common types for new APIs.

---
Refers to: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2023-01-01/storagecache.json:1838 in 30d4c64. [](commit_id = 30d4c64c2167a84401850e8f6f180f0aa0092b10, deletion_comment = False)

**Comment by mentat9** [comment]
@rebecca337 - I've signed off for ARM with comments.

**Comment by rebecca337** [comment]
@mentat9 Thanks for the note on v4, we are working on this and plan to have it in our next API version.

**Comment by rebecca337** [comment]
@tadelesh The CI-MissingBaseCommit tag is incorrect. If you look at our list of commits, we have the base commits in place done by OpenAPI.

**Comment by tadelesh** [comment]
> @tadelesh The CI-MissingBaseCommit tag is incorrect. If you look at our list of commits, we have the base commits in place done by OpenAPI.

I don't see the error you mentioned. But you need to fix the model validation error.

**Comment by rebecca337** [comment]
@tadelesh I pushed the fixes for this. The model validation is now failing incorrectly due to the following:

The required property primingManifestUrl is not present in GET or PUT responses. This is an issue with the validation checker since primingManifestUrl is labeled as both required and x-ms-secret. According to
the [x-ms-secret autorest specs](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret), "When a property is modeled as both "required" and "x-ms-secret": true, which means that this property must not exist in the response but has to be present in request." which is what we do in the examples.

"primingManifestUrl": {
 "description": "The URL for the priming manifest file to download. This file must be readable from the HPC Cache. When the file is in Azure blob storage the URL should include a Shared Access Signature (SAS) granting read permissions on the blob.",
 "$ref": "#/definitions/URLString",
 "x-ms-mutability": [
  "create"
 ],
 "x-ms-secret": true
},
"required": [
 "primingJobName",
 "primingManifestUrl"
],

**Comment by tadelesh** [comment]
> @tadelesh I pushed the fixes for this. The model validation is now failing incorrectly due to the following:
> 
> The required property primingManifestUrl is not present in GET or PUT responses. This is an issue with the validation checker since primingManifestUrl is labeled as both required and x-ms-secret. According to the [x-ms-secret autorest specs](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret), "When a property is modeled as both "required" and "x-ms-secret": true, which means that this property must not exist in the response but has to be present in request." which is what we do in the examples.
> 
> "primingManifestUrl": {  "description": "The URL for the priming manifest file to download. This file must be readable from the HPC Cache. When the file is in Azure blob storage the URL should include a Shared Access Signature (SAS) granting read permissions on the blob.",  "$ref": "#/definitions/URLString",  "x-ms-mutability": [   "create"  ],  "x-ms-secret": true }, "required": [  "primingJobName",  "primingManifestUrl" ],

Got it. I've let tooling team know the issue.

**Comment by rebecca337** [comment]
@tadelesh thanks. Are we approved to merge when we are ready? The other errors here are because of breaking change, and we already have the exception tag for that.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2023-01-01/storagecache.json [hunk 1]
```diff
@@ -0,0 +1,3511 @@
...
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/CacheNameParameter"
+          },
+          {
+            "description": "Object containing the user-selectable properties of the new Cache. If read-only properties are included, they must match the existing values of those properties.",
+            "in": "body",
```
Comment (tadelesh) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2023-01-01/storagecache.json:473]: Can we add `"required": true,` for this body param? The create or update action will be meaningless if there is no body param.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2023-01-01/storagecache.json [hunk 2]
```diff
@@ -0,0 +1,3511 @@
...
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/CacheNameParameter"
+          },
+          {
+            "$ref": "#/parameters/StorageTargetParameter"
+          },
+          {
+            "description": "Object containing the definition of a Storage Target.",
```
Comment (tadelesh) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2023-01-01/storagecache.json:1089]: Same issue with Cache_CreateOrUpdate.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2023-01-01/storagecache.json (hunk 1 above)
Comment (rebecca337) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2023-01-01/storagecache.json:473]: @tadelesh added this in latest commit

**Review by tadelesh — APPROVED** [review]

**Comment by rebecca337** [comment]
@tadelesh Could this PR be merged now? We are ready to merge.

**Comment by tadelesh** [comment]
@MaryGao @msyyc Please help to check the SDK breaking changes.

**Comment by rebecca337** [comment]
@tadelesh Are we missing anything on my team's end to get this merged? Our rollout is complete so we need to merge this Swagger change as soon as we can.

**Pull request merged**